daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -313,18 +334,33 @@ func (l *logStream) Close() error {
 
 // create creates log group and log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+	l.lock.RLock()
```

Comment:
This extra locking seems excessive.
Since in the case that there is a `create` in flight that is taking some time this will just block until it's done.
When it's open, the call is fast anyway so doesn't really need the extra level of locking.